sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RequestIdPolicy.java

```diff
@@ -3,24 +3,56 @@
 
 package com.azure.core.http.policy;
 
+import com.azure.core.http.HttpHeaders;
 import com.azure.core.http.HttpPipelineCallContext;
 import com.azure.core.http.HttpPipelineNextPolicy;
 import com.azure.core.http.HttpResponse;
 import reactor.core.publisher.Mono;
 
+import java.util.Objects;
 import java.util.UUID;
 
 /**
- * The pipeline policy that puts a UUID in the request header. Azure uses the request id as
+ * The pipeline policy that puts a UUID or user provided id in the request header. Azure uses the request id as
```

Comment:
With this change, we can't guarantee that the request-id is a unique identifier. User-provided request ids may not be unique.

Reply:
While it's true, it may also be possible that a customer generates their own UUID and may want to correlate it with the ID generated by their system.
I have no idea how the service implements it, but as a user I would expect the service to save the provided ID with the rest of the raw data I sent, so I can analyze it later. Therefore the SDK cannot guarantee uniqueness if user provides the ID, which is fine, if the persistence layer can support it. 
I want to suggest that the code will generate a default ID, and allow the customer to override it based on their need.